### ISSUE TITLE
Strange behavior when node is in hung/frozen state

### ISSUE THREAD

**Issue by andriyfomenko** [openshift/origin#9264]
OpenShift Origin cluster was originally deployed with openshift-ansible, over 3 VMs: 1 master, 2 nodes

I noticed that we are getting a lot of pods in "Terminating" state reported with "oc get pods"

An attempt to issue "oc delete pod ..." did not clean any of pods out

Reviewing this situation, I found that all "Terminating" pods were reported on node-2

Upon connecting to node-2 and issuing "docker ps", it became apparent that docker engine froze up for some reason: this command would lock terminal and hang indefinitely.

An attempt to restart docker also resulted in a freeze

Upon node restart, docker engine resumed normal operations, storage was not overflown, no other ill effects were apparent.

What appears to be a problem:

1. Node was not declared to be "dead" or "broken", "oc get nodes" reported it as "online/operational"
2. Pods were dead already on node-2, but were not cleaned u in the cluster
 
##### Version
$ oc version
oc v1.1.6
kubernetes v1.2.0-36-g4a3f9c5

##### Steps To Reproduce
Hard to reproduce, first time problem was observed

##### Current Result
Cluster operates normally after node-2 restart

##### Expected Result
Cluster to defect node and relevant pods

##### Additional Information
none

**Comment by ncdc** [comment]
Would you be able to upgrade to the latest 1.2 or 1.3 alpha? There are significant improvements in those versions related to pods getting stuck in Terminating status.

**Comment by andriyfomenko** [comment]
YEs, absolutely, the plan is to migrate to latest. This cluster is an "unstable development" for us and we would rebuild it i next 1-2 weeks anyway.

Please close the issue if you feel it is addressed in later version, I opened it mostly to flag the problem OpenShift had.

Thank you.
